### DISCUSSION TITLE
Fixing IndexError: cannot slice ListArray (of length 100000) with [...]

### DISCUSSION THREAD

**Discussion by Payal7890** [columnflow/columnflow#435] [Q&A] [answered]
I am seeking for help as I am really stuck in my analysis. I work on a repo which uses Columnflow. This is the repo: https://gitlab.cern.ch/dsavoiu/mttbar_trigger_study/-/tree/c4c513026defe16bb96f48798371421b42327719.
Every time I try to run the command: law run cf.SelectEvents --version v8 --selector default --workflow htcondor, I ran into an issue says: IndexError: cannot slice ListArray (of length 100000) with [...], which (to me) suggests that there is an attempt to access an index that is out of the valid range for a jagged array. Now I am at a loss which script to edit and where. The main default.py script is inside my area: /nfs/dust/cms/user/payalroy/mttbar_trigger_study/tgs/selection/default.py and this is the path to one output  log file: /nfs/dust/cms/user/payalroy/mttbar_trigger_study/data/tgs_store/analysis_tgs/cf.SelectEvents/run2_2017_nano_v9/tt_dl_powheg/nominal/calib__default/sel__default/v8/stdall_0To1.txt

I am also attaching one snapshot of the issue. It'd of great help if I get any lead.
<img width="1425" alt="Screenshot 2024-05-20 at 5 55 18 PM" src="https://github.com/columnflow/columnflow/assets/90550112/701ca992-d88c-495f-8479-efeb9ea5682d">

**Comment by mafrahm** [comment, marked as answer]
Hi @Payal7890, I converted this to a discussion since this is not a framework issue.

As far as I can see, you're trying to sort your Electron Collection via the Muon.pt, which does not work since the shapes won't match. 
Line 105 (should probably be `events.Electron.pt`):
```
electron_indices = sorted_indices_from_mask(electron_mask, events.Muon.pt)
```

**Comment by Payal7890** [comment]
Hello Mafrahm, thank you very much. Yes I am sorry. I got it.
